crates/agglayer-types/src/lib.rs

```diff
@@ -798,3 +969,53 @@ impl LocalNetworkStateData {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(transparent)]
+pub struct SettlementTxHash(Digest);
+
+impl SettlementTxHash {
+    pub const fn for_tests() -> Self {
+        SettlementTxHash(Digest::ZERO)
+    }
+
+    pub const fn new(hash: Digest) -> Self {
+        SettlementTxHash(hash)
+    }
+}
+
+impl From<Digest> for SettlementTxHash {
+    fn from(hash: Digest) -> Self {
+        SettlementTxHash(hash)
+    }
+}
+
+impl From<SettlementTxHash> for Digest {
+    fn from(tx_hash: SettlementTxHash) -> Self {
+        tx_hash.0
+    }
+}
+
+impl From<H256> for SettlementTxHash {
+    fn from(hash: H256) -> Self {
+        SettlementTxHash(Digest::from(*hash.as_fixed_bytes()))
+    }
+}
+
+impl From<SettlementTxHash> for H256 {
+    fn from(tx_hash: SettlementTxHash) -> Self {
+        tx_hash.0.as_bytes().into()
+    }
+}
+
+impl AsRef<Digest> for SettlementTxHash {
+    fn as_ref(&self) -> &Digest {
+        &self.0
+    }
+}
+
+impl fmt::Display for SettlementTxHash {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
```

Comment:
I believe these and a bunch of other impls in this file could be `derive_more`d.